vpr/src/place/net_cost_handler.h

```diff
@@ -25,135 +27,445 @@ enum class e_cost_methods {
+                                         std::vector<t_2D_bb>& bb_edge_new,
+                                         std::vector<t_2D_bb>& bb_coord_new);
+
+    /**
+     * @brief This function is called in update_layer_bb to update the net's bounding box incrementally if
+     * the pin under consideration is not changing layer.
+     * @param net_id ID of the net which the moving pin belongs to
+     * @param pin_old_loc Old location of the moving pin
+     * @param pin_new_loc New location of the moving pin
+     * @param curr_bb_edge The current known number of blocks of the net on bounding box edges
+     * @param curr_bb_coord The current known boudning box of the net
+     * @param bb_pin_sink_count_new The updated number of net's sinks on each layer
+     * @param bb_edge_new The new bb edge calculated by this function
+     * @param bb_coord_new The new bb calculated by this function
+     */
+     inline void update_bb_same_layer_(ClusterNetId net_id,
+                                             const t_physical_tile_loc& pin_old_loc,
+                                             const t_physical_tile_loc& pin_new_loc,
+                                             const std::vector<t_2D_bb>& curr_bb_edge,
+                                             const std::vector<t_2D_bb>& curr_bb_coord,
+                                             vtr::NdMatrixProxy<int, 1> bb_pin_sink_count_new,
+                                             std::vector<t_2D_bb>& bb_edge_new,
+                                             std::vector<t_2D_bb>& bb_coord_new);
+
+     double comp_per_layer_bb_cost_(e_cost_methods method);
+     double comp_cube_bb_cost_(e_cost_methods method);
+
+     /**
+     * @brief if "net" is not already stored as an affected net, add it in ts_nets_to_update.
+     * @param net ID of a net affected by a move
+     */
+     void record_affected_net_(const ClusterNetId net);
+
+     /**
+     * @brief To mitigate round-off errors, every once in a while, the costs of nets are summed up from scratch.
+     * This functions is called to do that for bb cost. It doesn't calculate the BBs from scratch, it would only add the costs again.
+     * @return Total bb (wirelength) cost for the placement
+     */
+     double recompute_bb_cost_();
+
+     /**
+     * @brief Given the 3D BB, calculate the wire-length cost of the net
+     * @param net_id ID of the net which cost is requested
+     * @param bb Bounding box of the net
+     * @return Wirelength cost of the net
+     */
+     double get_net_cube_bb_cost_(ClusterNetId net_id, const t_bb& bb);
+
+     /**
+     * @brief Given the per-layer BB, calculate the wire-length cost of the net on each layer
+     * and return the sum of the costs
+     * @param net_id ID of the net which cost is requested
+     * @param bb Per-layer bounding box of the net
+     * @return Wirelength cost of the net
+     */
+     double get_net_per_layer_bb_cost_(ClusterNetId /* net_id */,
```

Comment:
This commented out variable name is not necessary here I think. I think its only needed in the implementation to say that this variable goes unused. If you want to say that this variable goes unused, it may be best to put this in the `@param` above.